pandas/tests/io/json/test_json_table_schema.py

```diff
@@ -560,3 +560,13 @@ def test_multiindex(self, index_names):
         out = df.to_json(orient="table")
         result = pd.read_json(out, orient="table")
         tm.assert_frame_equal(df, result)
+
+    def test_empty_frame_roundtrip(self):
+        # GH 21287
+        df = pd.DataFrame([], columns=['a', 'b', 'c'])
+        expected = df.copy()
+        out = df.to_json(orient='table')
+        result = pd.read_json(out, orient='table')
+        # TODO: After DF coercion issue (GH 21345) is resolved, tighten type checks
+        tm.assert_frame_equal(expected, result,
```

Comment:
Minor nit but can you parametrize this with a "strict_check" parameter whose values can be `True` and `False`, with the former being marked as an xfail? You can see an example of this below:

https://github.com/pandas-dev/pandas/blob/0c65c57a279e755ab7093db925d1e580f9878dae/pandas/tests/io/json/test_json_table_schema.py#L498

The explicit xfail gives more visibility to the issue (I'm being overly cautious here)

Reply:
Sure, I’ll make the change. Have to say that I appreciate your pedantics on these! 😊